### PR TITLE
accept None as value for .set_page_background() + added documentation

### PR DESCRIPTION
**Pull request by semaeostomea**
as suggested in [this comment](https://github.com/PyFPDF/fpdf2/issues/487#issuecomment-1225306159) in #487 

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [x] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/PyFPDF/fpdf2/blob/master/LICENSE).

### CODE REVIEW

**Comment by semaeostomea** [comment]
Seeing as I'm struggling with the formatting of the code again, I'm wondering: how can I run these checks locally so I don't end up pushing commits for minor changes repeatedly all the time?

**Comment by RedShy** [comment]
> how can I run these checks locally so I don't end up pushing commits for minor changes repeatedly all the time?

Hi! For solving this issue, I have created the file `pre-commit` with this content
```bash
#!/bin/bash
git_cached_names() { git diff --cached --name-only --diff-filter=ACM; }
if git_cached_names | grep -q 'test.*\.py$' && grep -IRF generate=True $(git_cached_names | grep 'test.*\.py$'); then
    echo '`generate=True` left remaining in a call to assert_pdf_equal'
    exit 1
fi
modified_py_files=$(git_cached_names | grep '\.py$')
modified_fpdf_files=$(git_cached_names | grep '^fpdf.*\.py$')
# If any Python files were modified, format them:
if [ -n "$modified_py_files" ]; then
    if ! black --check $modified_py_files; then
        black $modified_py_files
        exit 1
    fi
    # If fpdf/ files were modified, lint them:
    [[ $modified_fpdf_files == "" ]] || pylint $modified_fpdf_files
fi
```
inside the folder `fpdf2/.git/hooks/`. In this way, every time I do a commit locally, it runs `black `and `pylint `on my code. 
I found the information [here](https://pyfpdf.github.io/fpdf2/Development.html#pre-commit-hook).

Hope it helps! :grin:

**Comment by semaeostomea** [comment]
> Hi! For solving this issue, I have created the file `pre-commit` with this content
> 
> ```shell
> #!/bin/bash
> git_cached_names() { git diff --cached --name-only --diff-filter=ACM; }
> if git_cached_names | grep -q 'test.*\.py$' && grep -IRF generate=True $(git_cached_names | grep 'test.*\.py$'); then
>     echo '`generate=True` left remaining in a call to assert_pdf_equal'
>     exit 1
> fi
> modified_py_files=$(git_cached_names | grep '\.py$')
> modified_fpdf_files=$(git_cached_names | grep '^fpdf.*\.py$')
> # If any Python files were modified, format them:
> if [ -n "$modified_py_files" ]; then
>     if ! black --check $modified_py_files; then
>         black $modified_py_files
>         exit 1
>     fi
>     # If fpdf/ files were modified, lint them:
>     [[ $modified_fpdf_files == "" ]] || pylint $modified_fpdf_files
> fi
> ```
> 
> inside the folder `fpdf2/.git/hooks/`. In this way, every time I do a commit locally, it runs `black `and `pylint `on my code. I found the information [here](https://pyfpdf.github.io/fpdf2/Development.html#pre-commit-hook).
> 
> Hope it helps! 😁

Thank you!
According to the documentation this is only for UNIX systems, so I'll probably have to do it manually (which I've been doing with black already, but not pylint)
But thank you for linking to the documentation, I didn't see there were develoopment guidelines with instructions to run the tests etc locally. This definitely answers my question 😊

**Comment by RedShy** [comment]
> According to the documentation this is only for UNIX systems

I suggest you to try anyway because I use Windows and it works too!

File: fpdf/fpdf.py [hunk 1]
```diff
@@ -1049,23 +1049,27 @@ def set_line_width(self, width):
...
+        ):
             if isinstance(background, drawing.DeviceRGB):
                 self.page_background = tuple(255 * v for v in background.colors)
             else:
                 self.page_background = background
         else:
             raise TypeError(
-                f"background must be of type str, io.BytesIO, PIL.Image.Image, drawing.DeviceRGB or tuple, got: {type(background)}"
+                f"""
+        background must be of type str, io.BytesIO, PIL.Image.Image, drawing.DeviceRGB, tuple or None
```
Comment (Lucas-C) [inline fpdf/fpdf.py:1071]:
```suggestion
                f"""background must be of type str, io.BytesIO, PIL.Image.Image, drawing.DeviceRGB, tuple or None
```

File: test/errors/test_FPDF_errors.py [hunk 2]
```diff
@@ -130,5 +130,7 @@ def test_invalid_page_background():
     with pytest.raises(TypeError) as error:
         pdf.set_page_background(i)
 
-    msg = f"background must be of type str, io.BytesIO, PIL.Image.Image, drawing.DeviceRGB or tuple, got: {type(i)}"
+    msg = f"""
+        background must be of type str, io.BytesIO, PIL.Image.Image, drawing.DeviceRGB, tuple or None
```
Comment (Lucas-C) [inline test/errors/test_FPDF_errors.py:134]:
```suggestion
    msg = f"""background must be of type str, io.BytesIO, PIL.Image.Image, drawing.DeviceRGB, tuple or None
```

File: fpdf/fpdf.py [hunk 3]
```diff
@@ -1049,23 +1049,27 @@ def set_line_width(self, width):
 
     def set_page_background(self, background):
         """
-        Sets a background color or image to be drawn every time `FPDF.add_page()` is called.
+        Sets a background color or image to be drawn every time `FPDF.add_page()` is called or removes a previously set background.
```
Comment (Lucas-C) [inline fpdf/fpdf.py:1052]:
```suggestion
        Sets a background color or image to be drawn every time `FPDF.add_page()` is called, or removes a previously set background.
```

**Review by Lucas-C — APPROVED** [review]

**Comment by semaeostomea** [comment]
@Lucas-C I've applied the changes. For some reason the codecov/project check is failing now though

**Comment by Lucas-C** [comment]
@semaeostomea: Merged, thank you for this extra PR!

**Pull request merged**